### PR TITLE
feat(reconciler): handle CR Not Found case

### PR DESCRIPTION
**Pull request by nissim-natanov-outreach**
## What this PR does / why we need it
Delegate Not Found errors to the handler via the new Deleted callback, and add unit test

**JIRA ID**: [QSS-822]

## Notes for your reviewers

[QSS-822]: https://outreach-io.atlassian.net/browse/QSS-822?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pkg/k8s/controllers/reconciler.go [hunk 1]
```diff
@@ -124,6 +137,11 @@ func (r *Reconciler) endReconcile(
 		return
 	}
 
+	if rr.Deleted {
```
Comment (evadnoob) [inline pkg/k8s/controllers/reconciler.go:140]: I think we should differentiate between "Not Found" or "other error", Not found could mean deleted.

Comment (nissim-natanov-outreach) [reply pkg/k8s/controllers/reconciler.go:140]: see below - r.Deleted is set ONLY on Not Found

Comment (nissim-natanov-outreach) [reply pkg/k8s/controllers/reconciler.go:140]: I also have a test for it - one for Not Found and another for 'any other err'

Comment (nissim-natanov-outreach) [reply pkg/k8s/controllers/reconciler.go:140]: I can rename to NotFound ...

**Review by george-e-shaw-iv — APPROVED** [review]
this looks fine just make sure you're being careful with your interface member functions and the interface in general. they're very hard to change once people have started using them unless you design them to be future-proof

if you pay attention in the stdlib there are barely any interfaces bigger than a few member functions. they do that intentionally and then make composite interfaces out of them

**Comment by nissim-natanov-outreach** [comment]
> this looks fine just make sure you're being careful with your interface member functions and the interface in general. they're very hard to change once people have started using them unless you design them to be future-proof
> 
> if you pay attention in the stdlib there are barely any interfaces bigger than a few member functions. they do that intentionally and then make composite interfaces out of them

agree - those are likely the most common cases. For other corner cases we would split logic into New interface and do type checks. Since DPO is the only operator using this for now, adding new core logic is fine.

File: pkg/k8s/controllers/reconciler.go (hunk 1 above)
Comment (nissim-natanov-outreach) [reply pkg/k8s/controllers/reconciler.go]: renamed Deleted to NotFound - although technically there is no case in which k8s would call Reconcile on never existing CRs :)

**Review by evadnoob — APPROVED** [review]
adds NotFound handing, this is in sync with the api to me, LGTM

**Pull request merged**